docs/README-develop-best-practices.md

```diff
@@ -0,0 +1,385 @@
+# Develoment Best Practices for cisco_node_utils APIs.
+
+* [Overview](#overview)
+* [YAML Development Best Practices](#ydbp)
+* [Common Object Development Best Practices](#odbp)
+* [MiniTest Development Best Practices](#mdbp)
+
+## <a name="overview">Overview</a>
+
+This document is intended to assist in developing cisco_node_utils API's that are consistent with current best practices.
+
+
+## <a name="ydbp">YAML Development Best Practices</a>
+
+* [Y1](#yaml1): All yaml feature entries should be kept in alphabetical order.
+* [Y2](#yaml2): Use anchors where needed for `config_get` and `config_get_token` entries.
+* Y3: Avoid nested optional matches.
+* [Y4](#yaml4): Use the `_template` feature when getting/setting the same value at multiple levels.
+* [Y5](#yaml5): When possible include a `default_value` that represents the system default value and document properties that don't have a system default. **Use strings rather then symbols when applicable**.
+* [Y6](#yaml6): When possible, use the same `config_get` show command for all properties and document any anomalies.
+* [Y7](#yaml7): Use Key-value wildcards instead of Printf-style wildcards.
+* Y8: Use of the `show running all` command is encouraged over `show <feature>` commands since not all `show <feature>` commands behave in the same manner across different cisco platforms.
```

Comment:
Can we recommend "show running <feature> all" as preferable over "show running all" if available?